JavaScript/JavaScriptSDK/TelemetryContext.ts

```diff
@@ -318,5 +320,67 @@ module Microsoft.ApplicationInsights {
                 }
             }
         }
+
+        private _fixDepricatedValues(envelope: Microsoft.ApplicationInsights.IEnvelope) {
+            try {
+                var data = (<any>envelope).data;
+                if (data && data.baseType === Microsoft.ApplicationInsights.Telemetry.RemoteDependencyData.dataType) {
+                    var rddData = <Microsoft.ApplicationInsights.Telemetry.RemoteDependencyData>(<any>data.baseData);
+                    if (rddData) {
+                        this._fixRDDDepricatedValues(rddData);
+                    }
+                }
+            } catch (e) {
+            }
```

Comment:
Consider logging a warning message. It won't be reported unless a user is in the debug mode or has enabled verboseLogging. 

sample code: 
                _InternalLogging.throwInternalNonUserActionable(LoggingSeverity.WARNING,
                    new _InternalLogMessage(_InternalMessageId.<newEnumName>, "error message" + Util.getExceptionName(e),
                        { exception: Util.dump(e) }));